docs/design/rubygems.md

```diff
@@ -0,0 +1,450 @@
+
+### Checksums
+
+Checksum validation is enabled by default.
+It can be disabled with the `BUNDLE_DISABLE_CHECKSUM_VALIDATION` environment variable.
+
+There is also an option to generate checksums in `Gemfile.lock`, but in bizarre way.
+(I have not found any other way to do it.)
+
+```shell
+# manually add `CHECKSUMS` section somewhere in the Gemfile.lock
+vim Gemfile.lock
+# install any gem
+bundle add rails --version "6.1.7"
+# check the Gemfile.lock /o\
+cat Gemfile.lock
+```
+
+Example of a checksum section in the `Gemfile.lock` file from my custom [repository](https://github.com/slimreaper35/cachi2-rubygems.git):
+
+```Gemfile.lock
+...
+DEPENDENCIES
+ rails (= 6.1.7)
+
+CHECKSUMS
+ actioncable (6.1.7) sha256=ee5345e1ac0a9ec24af8d21d46d6e8d85dd76b28b14ab60929c2da3e7d5bfe64
+ actionmailbox (6.1.7) sha256=c4364381e724b39eee3381e6eb3fdc80f121ac9a53dea3fd9ef687a9040b8a08
+ actionmailer (6.1.7) sha256=5561c298a13e6d43eb71098be366f59be51470358e6e6e49ebaaf43502906fa4
+ actionpack (6.1.7) sha256=3a8580e3721757371328906f953b332d5c95bd56a1e4f344b3fee5d55dc1cf37
+ actiontext (6.1.7) sha256=c5d3af4168619923d0ff661207215face3e03f7a04c083b5d347f190f639798e
+ actionview (6.1.7) sha256=c166e890d2933ffbb6eb2a2eac1b54f03890e33b8b7269503af848db88afc8d4
+ ...
+
+BUNDLED WITH
+ 2.5.11
+```
+
+I believe this feature is available since Bundler [v2.5.0](https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/lockfile_parser.rb#L55)
+from this [PR](https://github.com/rubygems/rubygems/pull/6374) being merged on Oct 21, 2023.
+
+### Environment variables
+
+The order of precedence for Bundler configuration options is as follows:
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environment variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+Since the local configuration takes higher precedence than the environment variables (except BUNLDE_APP_CONFIG),
+we need to set the Bundler configuration options to make the build work.
+We can easily set the environment variables if the local configuration file does not exist.
+
+#### Relevant environment variables
+
+```txt
+BUNDLE_FORCE_RUBY_PLATFORM=true
+BUNDLE_DEPLOYMENT=true
+BUNDLE_CACHE_PATH=${output_dir}/deps/rubygems
+```
+
+**BUNDLE_CACHE_PATH**: The directory that Bundler will place cached gems in when running bundle package,
+and that Bundler will look in when installing gems. Defaults to `vendor/cache`.
+
+**BUNDLE_DEPLOYMENT**: Disallow changes to the Gemfile.
+When the Gemfile is changed, and the lockfile has not been updated, running Bundler commands will be blocked.
+
+**BUNDLE_FORCE_RUBY_PLATFORM**: Ignore the current machine's platform and install only ruby platform gems.
+As a result, gems with native extensions will be compiled from source.
+
+See bundle config [documentation](https://bundler.io/v2.5/man/bundle-config.1.html).
+
+---
+
+Note: _BUNDLE_FORCE_RUBY_PLATFORM_ check is done via gemlock-parser in cachito.
```

Comment:
I was trying to double check this, but I think I got lost in the process. Can you point me to where we're setting this in the gemlock-parser?

Reply:
https://github.com/containerbuildsystem/cachito/blob/master/cachito/workers/pkg_managers/rubygems.py#L101